mock/src/lib.rs

```diff
@@ -110,6 +111,7 @@ pub fn consumed_note_data_ptr(note_idx: u32) -> memory::MemoryAddress {
 pub fn prepare_transaction(
     tx_inputs: TransactionInputs,
     tx_script: Option<TransactionScript>,
+    note_args: Option<BTreeMap<NoteId, Word>>,
     code: &str,
     file_path: Option<PathBuf>,
 ) -> PreparedTransaction {
```

Comment:
As I mentioned in one of the previous comments, I'm not sure we need to add `note_args` here. They can be fed directly to `execute_transactions()` method.